### PR TITLE
Extend Invoice reversal API to return affected ICs

### PR DESCRIPTION
**Pull request by pvpurcarcosmin**
https://github.com/metasfresh/metasfresh/issues/6827

### CODE REVIEW

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java [hunk 1]
```diff
@@ -573,6 +572,30 @@ public boolean isAvoidRecreate(final I_C_Invoice_Candidate ic)
...
+	}
+
+	@NonNull
+	public List<I_C_Invoice_Candidate> retrieveInvoiceCandidates(@NonNull final InvoiceId invoiceId)
+	{
+		final I_C_Invoice invoice = load(invoiceId, I_C_Invoice.class);
+
+		final IQuery<I_C_Invoice_Line_Alloc> invoiceLineAllocQuery =
+				queryBL.createQueryBuilder(I_C_InvoiceLine.class, invoice)
+						.addEqualsFilter(I_C_InvoiceLine.COLUMNNAME_C_Invoice_ID, invoice.getC_Invoice_ID())
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java:586]:
```suggestion
		final IQuery<I_C_Invoice_Line_Alloc> invoiceLineAllocQuery =
				queryBL.createQueryBuilder(I_C_InvoiceLine.class, invoice)
						.addEqualsFilter(I_C_InvoiceLine.COLUMNNAME_C_Invoice_ID, invoiceId)
```

* the equals filter can handle `RepoIdAware`s
* we don't need to pass the invoice a context provider; the query-builder can get the thread-local current transaction or create a "local" one on the fly

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java [hunk 2]
```diff
@@ -573,6 +572,30 @@ public boolean isAvoidRecreate(final I_C_Invoice_Candidate ic)
...
+						//collect invoice line alloc
+						.andCollectChildren(I_C_Invoice_Line_Alloc.COLUMN_C_InvoiceLine_ID)
+						.addOnlyActiveRecordsFilter()
+						.create();
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java:590]: it's OK like this but you *might* (didn't check) also use `andCollect` to get from `I_C_Invoice_Line_Alloc` to `I_C_Invoice_Candidate`
for next time maybe..

Comment (pvpurcarcosmin) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java:590]: Oh, just checked the method, that's pretty cool, I'll give it a try. thx!

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java (hunk 1 above)
Comment (pvpurcarcosmin) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandDAO.java:586]: I'll drop the loading of the invoice then, thx!

**Pull request merged**